gobblin-utility/src/main/java/gobblin/util/ClustersNames.java

```diff
@@ -112,6 +117,54 @@ private static String normalizeClusterUrl(String clusterIdentifier) {
     return clusterIdentifier;
   }
 
+  /**
+   *
+   * Returns the cluster name on which the application is running. Uses hadoop {@link Configuration} to get the url of
+   * the resourceManager or jobtracker. The URL is then translated into a human readable cluster name using
+   * {@link #getClusterName(String)}
+   *
+   * <p>
+   * <b>MapReduce mode</b> Uses the value for "yarn.resourcemanager.address" from {@link Configuration} excluding the
+   * port number. If "yarn.resourcemanager.address" is not set, (possible in Hadoop1), falls back to
+   * "mapreduce.jobtracker.address"
+   * </p>
+   *
+   * <p>
+   * <b>Standalone mode (outside of hadoop)</b> Uses the Hostname of {@link InetAddress#getLocalHost()}
+   * </p>
+   *
+   * <p>
+   * Use {@link #getClusterName(String)} if you already have the cluster URL
+   * </p>
+   *
+   */
+  public String getClusterName() {
```

Comment:
We should make this take a Configuration as argument (in case not using default configuration).